test/test_torch.py

```diff
@@ -5949,6 +5949,10 @@ def helper(dim, dtype, device, size_result, size_source):
                 index = (torch.ones(256) * 257).to(dtype=torch.long)
                 self.assertRaises(RuntimeError, lambda: result.index_add_(dim, index, source))
 
+        source = torch.tensor([2., 4.])
+        result = torch.tensor([[1., 2.], [4., 5.], [7., 8.]])
+        self.assertRaises(RuntimeError, lambda: result.index_add_(0, torch.tensor([0, 2]), source))
+
```

Comment:
`test_torch.py` is legacy. Please add an error input the relevant OpInfo file in `torch/testing/_internal/common_methods_invocations.py`

Reply:
Just moved it over, please let me know if I should iterate on my changes.